caffe2/serialize/boost_crc.h

```diff
@@ -0,0 +1,1099 @@
+//  Boost CRC library crc.hpp header file  -----------------------------------//
```

Comment:
Maybe this should go in the relevant third_party dir? (i.e. fbcode/caffe2/third_party)

Also, if this is mostly for boosting the sw crc speed, the complexity of this dependency might not be warranted (given that we probably only want to run the sw version on a few of the last/first unaligned bytes on the buffer). I suspect that if adding these is green-lit, we might need to do things like including the LICENSE file (see line 4 below), etc.

Reply:
Yeah. will move to third_party!

I think there may be cases when it runs only the SW version? I do not know yet. @suo thoughts?

Reply:
Either way, I plan to have both HW and SW support here.